README.md

```diff
@@ -1,28 +1,204 @@
+
+#### Factory
+##### Usage
+
 ```js
-{
-  url: '/foo/123?bar=baz',
-  path: '/foo/123',
-  params: {id: 123},
-  query: {bar: 'baz'}
-}
+var urlMapper = require('url-mapper');
+var mapper = urlMapper(options);
 ```
 
-This library just allows you to map a url to a function. It allows dynamic segments and you can pass a full url or just the path. It's designed to be used with other libraries to manage URLs in your applications.
+##### Arguments
+
+<table>
+  <tr>
+    <th>Param</th><th>Type</th><th>Details</th>
+  </tr>
+  <tr>
+    <td>options</th><td><code>Object</code></td>
+    <td>Options passed to converter.
+      <table>
+        <tr>
+          <th>Property</th><th>Type</th><th>Details</th>
+        </tr>
+        <tr>
+          <td>query</td><td><code>Boolean</code></td>
+          <td>Enables converting values not defined in route as query in URL Object Notation</td>
+        </tr>
+      </table>
+    </td>
+  </tr>
+</table>
+
+##### Returns
+
+`Object` - Object with `parse`, `stringify` and `map` methods.
+
+Returned methods deals with Express-style route definitions and cleaned routable part of url (without origin, base path, leading hash symbol).
+
+Params defined in route is mapped to the same named properties in `values` Object with help of `path-to-regexp` module.
+It safe to pass Numbers and Booleans as well as Strings as path parameteres.
+Original type would be preserved while parsing back stringified one.
+
+By default, query part is ignored.
+Query part params is mapped to the same named properties in `values` Object if `{ query: true }` option was passed to factory.
+Conversion for query part is made with help of `URLON` module. Therefore, it can accept any JSON serializable value.
+
+Hash part is ignored at all if any present.
+You still can manage your routes in `location.hash` but don't provide `#` symbol before routable part.
+
+#### `parse` method
+
+##### Usage
+
+`mapper.parse(route, url)`;
+
+##### Arguments
+
+Param | Type     | Details
+------|----------|--------
+route | `String` | Express style route definition
+url   | `String` | Routable part of url
+
+##### Returns
+
+`Object` - values parsed from `url` with given `route`.
+
+Path parsed using `path-to-regexp` module, tweaked to support `Boolean` and `Number`.
+Query part parsed with `URLON` module if it enabled in factory.
+
+#### `stringify` method
+
+##### Usage
+
+`mapper.stringify(route, values)`;
+
+##### Arguments
+
+Param  | Type     | Details
+-------|----------|--------
+route  | `String` | Express style route definition
+values | `Object` | Object used to populate parameters to url
+
+##### Returns
+
+`String` - values stringified to `url` with given `route`.
+
+Properties defined at route is stringified to path part using `path-to-regexp` module, tweaked to support `Boolean` and `Number`.
+Properties not defined at route is stringified to query part using `URLON` module if it enabled in factory.
```

Comment:
Properties not defined in route are stringified .. ... if { query: true } option was passed to factory.